pandas/tests/tools/test_concat.py

```diff
@@ -1899,3 +1899,38 @@ def test_concat_multiindex_dfs_with_deepcopy(self):
         tm.assert_frame_equal(result_copy, expected)
         result_no_copy = pd.concat(example_dict, names=['testname'])
         tm.assert_frame_equal(result_no_copy, expected)
+
+
+    def test_concat_no_unnecessary_upcats(self):
```

Comment:
I need `pytest` module for using `@parametrize`.
Should I create another class for pytest?